contracts/zero-ex/contracts/src/features/multiplex/BatchMultiplexFeature.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+  Copyright 2024 ZeroEx Intl., Rigo Intl.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/v06/LibSafeMathV06.sol";
+import "../../examples/BatchMultiplexValidator.sol";
+import "../../fixins/FixinCommon.sol";
+import "../../fixins/FixinReentrancyGuard.sol";
+import "../../migrations/LibMigrate.sol";
+import "../interfaces/IFeature.sol";
+import "../interfaces/IBatchMultiplexFeature.sol";
+
+/// @dev This feature enables batch transactions by re-routing the single swaps to the exchange proxy.
+contract BatchMultiplexFeature is IFeature, IBatchMultiplexFeature, FixinCommon, FixinReentrancyGuard {
+    bytes4 private constant _VALIDATE_SELECTOR = BatchMultiplexValidator.validate.selector;
+
+    // TODO: remove as this is mock for validator setup and tests
+    address public immutable _validator;
+
+    /// @inheritdoc IFeature
+    string public constant override FEATURE_NAME = "BatchMultiplexFeature";
+    /// @inheritdoc IFeature
+    uint256 public immutable override FEATURE_VERSION = _encodeVersion(1, 0, 0);
+
+    /// @dev Refunds up to `msg.value` leftover ETH at the end of the call.
+    modifier refundsAttachedEth() {
+        _;
+        uint256 remainingBalance = LibSafeMathV06.min256(msg.value, address(this).balance);
```

Comment:
would this still be valid if the proxy had some eth prior to the call?
if someone sends eth to the proxy by mistake for instance

Reply:
yes, this method will refund a maximum of the amount sent, so if the proxy already holds ETH, it won't withdraw what was already in the proxy. This means that if ETH is accidentally attached to the batchMultiplex call, it will be refunded. If, on the other hand, eth was received before the call, the batchMultiplex method won't be able to transfer said ETH.

Reply:
if the proxy had eth before the call, and the amount is more than `msg.value` this will refund `msg.value` even if the swap has been successful no? bc it will be taken from previous eth

Reply:
yes, you are right! I think it would work also in the case `msg.value` is higher than the previous balance and sent eth is spent in the sub-calls, the proxy balance would be the previous balance plus any leftover, and all would be refunded. Technically this would violate the `onlyOwner` modifier of [FundRecoveryFeature](https://github.com/RigoBlock/0x-protocol/blob/e66307ba319e8c3e2a456767403298b576abc85e/contracts/zero-ex/contracts/src/features/FundRecoveryFeature.sol#L43). I am open to modify the assertion by calculating the proxy eth balance before the execution and refunding the difference. This would add a couple more if conditions, and I am happy to brainstorm more on this.